homeassistant/components/vlc_telnet/media_player.py

```diff
@@ -309,16 +310,21 @@ async def async_play_media(
             media_id = sourced_media.url
 
         # Sign and prefix with URL if playing a relative URL
-        if media_id[0] == "/":
-            media_id = async_sign_path(
-                self.hass,
-                quote(media_id),
-                timedelta(seconds=media_source.DEFAULT_EXPIRY_TIME),
-            )
+        if media_id[0] == "/" or is_hass_url(self.hass, media_id):
+            parsed = yarl.URL(media_id)
+
+            if parsed.query:
+                LOGGER.debug("Not signing path for content with query param")
+            else:
+                media_id = async_sign_path(
+                    self.hass,
+                    quote(media_id),
+                    timedelta(seconds=media_source.DEFAULT_EXPIRY_TIME),
+                )
 
             # prepend external URL
-            hass_url = get_url(self.hass)
-            media_id = f"{hass_url}{media_id}"
+            if media_id[0] == "/":
+                media_id = f"{get_url(self.hass)}{media_id}"
 
         if media_type != MEDIA_TYPE_MUSIC:
```

Comment:
We should probably move this up before signing and prefixing the URL.

Reply:
done.